src/blockstore/gateway.ts

```diff
@@ -17,17 +19,17 @@ export type UnsubscribeResult = Result<() => void>;
 export interface Gateway {
   // all the methods never throw!
   // an error is reported as a Result
-  buildUrl(baseUrl: URI, key: string): Promise<Result<URI>>;
+  buildUrl(baseUrl: URI, key: string, loader?: Loadable): Promise<Result<URI>>;
```

Comment:
why is loader not set in the constructor?

Reply:
is it b/c they wrap each other at call time?

Reply:
and as you see it's optional it's build for the key attachment thing